src/definitions/definitionBlocks.ts

```diff
@@ -192,16 +221,40 @@ export interface InputDefinitionBlock<TypeName extends string> extends NexusGenC
 
 export class InputDefinitionBlock<TypeName extends string> {
   readonly typeName: string
-  constructor(protected typeBuilder: InputDefinitionBuilder, protected isList = false) {
+  constructor(
+    protected typeBuilder: InputDefinitionBuilder,
+    protected nonNullDefault: boolean,
```

Comment:
Why do we need to pass `notNullDefault` through here?

Reply:
Because we need the nonNullDefault to be passed to the wrapping function.. That function needs the nonNull Default to properly add the `nonNull` if it's nonNull by default

Reply:
Could we source that later in the building phase when the wrapping occurs, or did something change here that I'm missing compared to the previous behavior?

Reply:
So what's technically changed is that previously we were just setting a config (list:true) vs. now it's the actual type that we need to change.

Couple solutions:

- We get the `wrapping` out of the definition block thanks to the typeBuilder, and we apply wrap the type outside of it

- It's a bit dirty but we actually don't need to apply the proper nonNullDefault at this stage, since we'll need to unwrap/rewrap with graphql types later anyway, and with this PR we're now applying the nonNullDefaults when wrapping.